hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DBScanner.java

```diff
@@ -305,7 +335,7 @@ private void processRecords(ManagedRocksIterator iterator,
   }
 
   private boolean withinLimit(long i) {
-    return limit == -1L || i < limit;
+    return recordsPerFile > 0 || limit == -1L || i < limit;
```

Comment:
if the `recordsPerFile > 0` is `true`, the subsequent judgments will be short-circuited, including the `i < limit` then the limit will be invalidated. This is not a expected function.

Reply:
Thanks @xichen01  for the comment and review.
When `recordsPerFile>0`, it means that `--max-records-per-file` has taken effect, and `--limit` should be ignored at this time.

Reply:
The `--limit` is used to limit the total count of records, the `--max-records-per-file` is used to limit the max records count of specific file.

Such as: 
` ozone debug ldb ...  --limit 10 --max-records-per-file 1 --out result.txt`

This command should generate 10 files, like result.txt0, result.txt1, ..., and each of them has 1 record.

Reply:
I'll update later.